src/Bicep.Core/Registry/Oci/OciArtifactReference.cs

```diff
@@ -70,14 +70,18 @@ public class OciArtifactReference : ArtifactReference, IOciArtifactReference
         public override bool IsExternal => true;
 
         // unqualifiedReference is the reference without a scheme or alias, e.g. "example.azurecr.invalid/foo/bar:v3"
-        public static ResultWithDiagnostic<OciArtifactReference> TryParseModule(string? aliasName, string unqualifiedReference, RootConfiguration configuration, Uri parentModuleUri)
+        // The configuration and parentModuleUri are needed to resolve aliases and experimental features
+        public static ResultWithDiagnostic<OciArtifactReference> TryParseModuleAndAlias(string? aliasName, string unqualifiedReference, RootConfiguration configuration, Uri parentModuleUri)
             => TryParse(ArtifactType.Module, aliasName, unqualifiedReference, configuration, parentModuleUri);
 
-        public static ResultWithDiagnostic<OciArtifactReference> TryParse(ArtifactType type, string? aliasName, string unqualifiedReference, RootConfiguration configuration, Uri parentModuleUri)
+        public static ResultWithDiagnostic<OciArtifactReference> TryParseModule(string unqualifiedReference)
+            => TryParse(ArtifactType.Module, null, unqualifiedReference, null, null);
+
+        public static ResultWithDiagnostic<OciArtifactReference> TryParse(ArtifactType type, string? aliasName, string unqualifiedReference, RootConfiguration? configuration, Uri? parentModuleUri)
         {
             if (TryParseParts(type, aliasName, unqualifiedReference, configuration).IsSuccess(out var parts, out var errorBuilder))
             {
-                return new(new OciArtifactReference(type, parts.Registry, parts.Repository, parts.Tag, parts.Digest, parentModuleUri));
+                return new(new OciArtifactReference(type, parts.Registry, parts.Repository, parts.Tag, parts.Digest, parentModuleUri ?? new Uri("file:///no-parent-file-is-available.bicep")));
```

Comment:
I think that the scenarios that invoke `TryParseModule` should pass a `parentModuleUri` so that you are not forced to create this argument.
In the first reference (constructor of `SourceArchive`), isn't the parentModuleUri described in the file `ArchiveMetadata.Entrypoint` ?
In the second reference (`BicepExternalSourceDocumentLinkHandler`) shouldn't the incomming request payload (`ExternalSourceDocumentLinkData`) contain the `parentModuleUri` ?